### PR TITLE
[TEST] Put options that enable assertions earlier on command line

### PR DESCRIPTION
**Pull request by droberts195**
This change moves the -ea and -esa options that enable assertions for
test nodes before the cluster-specific JVM arguments on the Java command
line.  This opens up the possibility for the cluster-specific JVM
arguments to disable assertions for one particular package or class,
which can be useful in BWC testing where incorrect assertions cannot be
removed from released versions of the product.

### CODE REVIEW

**Review by jaymode — APPROVED** [review]
LGTM

**Comment by jasontedor** [comment]
@droberts195 reached out via another channel and clarified the intent of this is to handle nodes that on older versions for which assertions trip during tests. With this change, we can disable the specific assertions that are failing and continue to run our BWC tests against those older nodes. This is a good change.

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy [hunk 1]
```diff
@@ -167,7 +167,10 @@ class NodeInfo {
         String collectedSystemProperties = config.systemProperties.collect { key, value -> "-D${key}=${value}" }.join(" ")
         String esJavaOpts = config.jvmArgs.isEmpty() ? collectedSystemProperties : collectedSystemProperties + " " + config.jvmArgs
         if (Boolean.parseBoolean(System.getProperty('tests.asserts', 'true'))) {
-            esJavaOpts += " -ea -esa"
+            // put the enable assertions options before other options to allow
+            // flexibility to disable assertions for specific packages or classes
+            // in the cluster-specific options
+            esJavaOpts = "-ea -esa " + esJavaOpts
```
Comment (jasontedor) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy]: How about:

```diff
diff --git a/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy b/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy
index 693d62679a..686233bdfe 100644
--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy
@@ -170,7 +170,7 @@ class NodeInfo {
             // put the enable assertions options before other options to allow
             // flexibility to disable assertions for specific packages or classes
             // in the cluster-specific options
-            esJavaOpts = "-ea -esa " + esJavaOpts
+            esJavaOpts = String.join(" ", "-ea", "-esa", esJavaOpts)
         }
         env.put('ES_JAVA_OPTS', esJavaOpts)
         for (Map.Entry<String, String> property : System.properties.entrySet()) {
```

**Review by jasontedor — APPROVED** [review]
LGTM. I left a minor suggestion.

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy (hunk 1 above)
Comment (droberts195) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy]: Thanks.  I implemented the suggestion.

**Pull request merged**